sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -68,7 +68,7 @@ class ResolveSessionCatalog(val catalogManager: CatalogManager)
       throw QueryCompilationErrors.unsupportedTableOperationError(ident, "REPLACE COLUMNS")
 
     case a @ AlterColumn(ResolvedTable(catalog, ident, table: V1Table, _), _, _, _, _, _, _)
-        if isSessionCatalog(catalog) =>
+        if isSessionCatalog(catalog) && !isV2SessionCatalog(catalog) =>
```

Comment:
maybe we should only have one method `supportsV1Command(catalog)`, and it returns true only if the catalog name is `spark_catalog` and there is no custom catalog to override it.